test-clients/src/main/java/com/hedera/services/bdd/suites/token/UniqueTokenManagementSpecs.java

```diff
@@ -92,46 +99,243 @@ public static void main(String... args) {
+						getTokenNftInfosFailsWithDeletedTokenNft()
+        );
+    }
+
+	private HapiApiSpec populatingMetadataForFungibleDoesNotWork() {
+		return defaultHapiSpec("PopulatingMetadataForFungibleDoesNotWork")
+				.given(
+						newKeyNamed(SUPPLY_KEY),
+						cryptoCreate(TOKEN_TREASURY),
+						tokenCreate(FUNGIBLE_TOKEN)
+								.initialSupply(0)
+								.tokenType(TokenType.FUNGIBLE_COMMON)
+								.supplyType(TokenSupplyType.INFINITE)
+								.supplyKey(SUPPLY_KEY)
+								.treasury(TOKEN_TREASURY)
+				).when(
+						mintToken(FUNGIBLE_TOKEN, List.of(
+								metadata("some-data"),
+								metadata("some-data2"),
+								metadata("some-data3"),
+								metadata("some-data4")
+						)).hasKnownStatus(INVALID_TOKEN_MINT_AMOUNT).via("should-not-work")
+				).then(
+						getAccountBalance(TOKEN_TREASURY).hasTokenBalance(FUNGIBLE_TOKEN, 0),
+						getTxnRecord("should-not-work").showsNoTransfers(),
+						UtilVerbs.withOpContext((spec, opLog) -> {
+							var mintNFT = getTxnRecord("should-not-work");
+							allRunFor(spec, mintNFT);
+							var receipt = mintNFT.getResponseRecord().getReceipt();
+							Assert.assertEquals(0, receipt.getNewTotalSupply());
+						})
+				);
+	}
+
+	private HapiApiSpec populatingAmountForNonFungibleDoesNotWork() {
+		return defaultHapiSpec("PopulatingAmountForNonFungibleDoesNotWork")
+				.given(
+						newKeyNamed(SUPPLY_KEY),
+						cryptoCreate(TOKEN_TREASURY),
+						tokenCreate(NFT)
+								.initialSupply(0)
+								.tokenType(NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.supplyKey(SUPPLY_KEY)
+								.treasury(TOKEN_TREASURY)
+				).when(
+						mintToken(NFT, 300).hasKnownStatus(INVALID_TOKEN_MINT_METADATA).via("should-not-work")
+				).then(
+						getTxnRecord("should-not-work").showsNoTransfers(),
+						getAccountBalance(TOKEN_TREASURY).hasTokenBalance(NFT, 0),
+						UtilVerbs.withOpContext((spec, opLog) -> {
+							var mintNFT = getTxnRecord("should-not-work");
+							allRunFor(spec, mintNFT);
+							var receipt = mintNFT.getResponseRecord().getReceipt();
+							Assert.assertEquals(0, receipt.getNewTotalSupply());
+							Assert.assertEquals(0, receipt.getSerialNumbersCount());
+						})
+				);
+	}
+
+	private HapiApiSpec fungibleCommonMaxSupplyReachWork() {
+		return defaultHapiSpec("FungibleCommonMaxSupplyReachWork")
+				.given(
+						newKeyNamed(SUPPLY_KEY),
+						cryptoCreate(TOKEN_TREASURY).balance(10L),
+						tokenCreate(FUNGIBLE_TOKEN)
+								.initialSupply(0)
+								.maxSupply(500)
+								.tokenType(TokenType.FUNGIBLE_COMMON)
+								.supplyType(TokenSupplyType.FINITE)
+								.supplyKey(SUPPLY_KEY)
+								.treasury(TOKEN_TREASURY)
+				)
+				.when(
+						mintToken(FUNGIBLE_TOKEN, 3000).hasKnownStatus(TOKEN_MAX_SUPPLY_REACHED).via("should-not-appear")
+				)
+				.then(
+						getTxnRecord("should-not-appear").showsNoTransfers(),
+						getAccountBalance(TOKEN_TREASURY).hasTinyBars(10L),
```

Comment:
This check is not necessary because our test has nothing to do with hbars. We can check the token balance though